x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_types/eql/eql.ts

```diff
@@ -165,8 +165,21 @@ export const eqlExecutor = async ({
 
       let newSignals: Array<WrappedFieldsLatest<BaseFieldsLatest>> | undefined;
 
+      // @ts-expect-error shard_failures exists in
+      // elasticsearch response v 8.18.x
+      // TODO: remove ts-expect-error when ES lib version is updated
+      const shardFailures = response.shard_failures;
+
       const { events, sequences } = response.hits;
 
+      if (shardFailures) {
+        const shardFailureMessage = `The EQL event query was only executed on the available shards. The query failed to run successfully on the following shards: ${JSON.stringify(
```

Comment:
We should internationalize this copy.

Reply:
Can we do that server-side now? I didn't think we could..

Reply:
Added i18n!!